File: jar/src/main/java/jline/api/sn/SnHasMixedClasses.java
package jline.api.sn;

import jline.lang.NetworkStruct;

public final class SnHasMixedClasses {
    private SnHasMixedClasses() {}

    /**
     * Checks if the network has both open and closed classes
     *
     * @param sn NetworkStruct object for the queueing network model
     * @return boolean
     */
    public static boolean snHasMixedClasses(NetworkStruct sn) {
        return sn.njobs.hasFinite() && sn.njobs.hasInfinite();
    }
}

File: jar/src/main/kotlin/jline/api/sn/SnHasMixedClasses.kt
package jline.api.sn

import jline.lang.NetworkStruct

/**
 * Checks if the network has both open and closed classes
 *
 * @param sn - NetworkStruct object for the queueing network model
 * @return boolean
 */
fun snHasMixedClasses(sn: NetworkStruct): Boolean {
    return sn.njobs.hasFinite() && sn.njobs.hasInfinite()
}
/**
 * Stochastic network HasMixedClasses algorithms
 */
@Suppress("unused")
class SnhasmixedclassesAlgo {
    companion object {
        // Class documentation marker for Dokka
    }
}